### DISCUSSION TITLE
Documentation between "device" and "application": How-to?

### DISCUSSION THREAD

**Discussion by hbswn** [Seeed-Studio/wiki-documents#699] [Ideas]
After writing [/grove-sgp41-with-aht20/](https://wiki.seeedstudio.com/grove-sgp41-with-aht20/) I realized it's _**not**_ describing an "application": it's nowhere near anything an end-user could use _and_ it should not be.

So what's the value: it describes a **concept**, that should be used in real-life applications.
( In this specific case, because using the defaults only produces correct raw values at 25°C and 50% relative humidity.)

I think it makes sense to add a separate category for this kind of documentation. The sensor and micro-controller device documentation probably contains similar cases (in the examples).

A good example from the early Linux world are the [HOWTOs](https://tldp.org/HOWTO/HOWTO-INDEX/howtos.html).

HOWTOs:
- describe a single concept or a few closely related concepts
- describe what you want to achieve, (usually) instead of a specific device
- provide working/tested examples
- link to resources (hardware data, source code, more detailed documentation, use cases, . . .)

Or maybe such a category exists and I haven't found it, yet ?

> UPDATE: I've found some under [Arduino->Tutorials](https://wiki.seeedstudio.com/Arduino/)

regards,

Hans

**Comment by MatthewJeffson** [comment]
Yes, and that is indeed an interesting idea. I will look into it!
Thanks!
Matthew

**Comment by hbswn** [comment]
User Stories may be the best to add: solving a collection of User Stories may result in an application. The idea is that the stories are well defined, not too big and verifiable.

Some links:
- [Wikipedia: User story](https://en.wikipedia.org/wiki/User_story)
- [Scrum.org: User Story or Stakeholder Story?](https://www.scrum.org/resources/blog/user-story-or-stakeholder-story)
- [Tech at GSA: Writing Effective User Stories](https://tech.gsa.gov/guides/effective_user_stories/)

The concept is mostly used in "agile" software development, but it was first used in a car company so I think it will work here as well.

Hans

**Reply by MatthewJeffson** [reply]
That is an interesting idea, and we indeed need more topics to help both users and contributors.

**Comment by hbswn** [comment]
It's more about the user perspective here: think about how the device or software is used.

"User" is quite a broad term in this approach: often it's just another application or device's requirement.
